src/quickpicks/utils/quickPickUtils.ts

```diff
@@ -0,0 +1,251 @@
+   * ```
+   */
+  selectedItems?: T[];
+
+  /**
+   * Controls whether the QuickPick should close when focus is lost.
+   * When true, the QuickPick stays open even when focus moves elsewhere.
+   * @default false
+   */
+  ignoreFocusOut?: boolean;
+
+  /**
+   * Callback triggered when the user accepts the current selection.
+   * This occurs when the user presses Enter or clicks the OK button.
+   *
+   * @example
+   * ```typescript
+   * onDidAccept: async (quickPick) => {
+   *   const selectedItems = quickPick.selectedItems;
+   *   await processSelection(selectedItems);
+   *   quickPick.hide();
+   * }
+   * ```
+   */
+  onDidAccept?: (quickPick: vscode.QuickPick<T>) => Promise<void> | void;
+}
+
+/**
+ * Creates and returns a QuickPick instance with enhanced functionality.
+ * The QuickPick will be shown immediately and the function will wait for completion.
+ *
+ * @param items The items to show in the QuickPick
+ * @param options Enhanced QuickPick options
+ * @returns A Promise that resolves to an object containing the QuickPick instance and selected items after it is hidden
+ */
+export function createEnhancedQuickPick<T extends QuickPickItemWithValue<any>>(
+  items: T[] | Promise<T[]>,
+  options?: EnhancedQuickPickOptions<T>,
+): Promise<{ quickPick: vscode.QuickPick<T>; selectedItems: T[] }> {
+  const quickPick = vscode.window.createQuickPick<T>();
+  let selectedItems: T[] = [];
+
+  // Set standard options
+  if (options) {
+    quickPick.placeholder = options.placeHolder;
+    quickPick.ignoreFocusOut = options.ignoreFocusOut ?? false;
+    quickPick.title = options.title;
+    quickPick.canSelectMany = options.canSelectMany ?? false;
+    quickPick.matchOnDescription = options.matchOnDescription ?? false;
+    quickPick.matchOnDetail = options.matchOnDetail ?? false;
+    quickPick.buttons = options.buttons ?? [];
+  }
+
+  // Set items (handle promise if needed)
+  if (!(items instanceof Promise)) {
+    quickPick.items = items;
+  } else {
+    // Handle promise asynchronously
+    (async () => {
+      quickPick.busy = true;
+      quickPick.items = await items;
+      quickPick.busy = false;
+
+      // Set default selected items if provided (after items are loaded)
+      if (options?.selectedItems && options.selectedItems.length > 0) {
+        quickPick.selectedItems = options.selectedItems;
+      }
+    })();
+  }
+
+  // Set default selected items if provided and items are already available
+  if (!(items instanceof Promise) && options?.selectedItems && options.selectedItems.length > 0) {
+    quickPick.selectedItems = options.selectedItems;
+  }
+
+  // Set up event handlers
+  if (options?.onSelectionChange) {
+    quickPick.onDidChangeSelection((items: readonly T[]) => {
+      options.onSelectionChange?.(items, quickPick);
+    });
+  }
+
+  if (options?.onActiveItemChange) {
+    quickPick.onDidChangeActive((items: readonly T[]) => {
+      options.onActiveItemChange?.(items[0], quickPick);
+    });
+  }
```

Comment:
One other thing I noticed: let's make sure we're keeping track of these VS Code `Disposable`s (and the others below this block):
```suggestion
  const disposables: Disposable[] = [];
  
  // Set up event handlers
  if (options?.onSelectionChange) {
    disposables.push(
      quickPick.onDidChangeSelection((items: readonly T[]) => {
        options.onSelectionChange?.(items, quickPick);
      })
    );
  }

  if (options?.onActiveItemChange) {
    disposables.push(
      quickPick.onDidChangeActive((items: readonly T[]) => {
        options.onActiveItemChange?.(items[0], quickPick);
      });
    )
  }
```
(This is new since the code on `main` wasn't properly cleaning them up, but we'll want to include this so other quickpicks using this won't have to worry about listener cleanup.)